packages/input/src/Input/Input.tsx

```diff
@@ -0,0 +1,157 @@
+};
+
+type RootProps = BaseRootProps &
+  Omit<React.ComponentPropsWithoutRef<"input">, "size">;
+
+type RootRef = HTMLInputElement;
+
+type InternalProps = Omit<BaseRootProps, "error"> & {
+  state: "default" | "disabled" | "error";
+};
+
+type Required<T> = {
+  [P in keyof T]-?: T[P];
+};
+
+const InputContext = React.createContext<Required<InternalProps>>({
+  state: "default",
+  size: "2",
+  variant: "outline",
+});
+
+const Root = React.forwardRef<RootRef, RootProps>(
+  (
+    {
+      size = "2",
+      variant = "outline",
+      className,
+      disabled,
+      error,
+      children,
+      ...props
+    },
+    forwardedRef,
+  ) => {
+    const inputRef = React.useRef<HTMLInputElement>(null);
+    const composedRefs = useComposedRefs(forwardedRef, inputRef);
+
+    const state = disabled ? "disabled" : error ? "error" : "default";
+
+    return (
+      <InputContext.Provider value={{ size, variant, state }}>
+        <div
+          className={clsx(
+            "box-border flex items-center transition-all",
+            "border-2 border-solid text-gray-12 placeholder:text-gray-10",
+            COLOR.Container[state][variant],
+            SIZE.Container[size],
+          )}
+          // Focus the input when clicking on the container
+          onPointerDown={(event) => {
+            const target = event.target as HTMLElement;
+
+            // Make sure we're not clicking on an interactive element
+            if (target.closest("button, a")) {
+              event.preventDefault();
+              return;
+            }
+
+            const input = inputRef.current;
+            if (!input) return;
+
+            requestAnimationFrame(() => {
+              input.focus();
+            });
+          }}
+        >
+          <input
+            className={clsx(
+              "appearance-none text-gray-12 border-none shadow-0 outline-0 bg-transparent",
+              "[font-family:inherit] h-full w-full",
+              "order-2",
+              SIZE.Input[size],
+              className,
+            )}
+            disabled={disabled}
+            {...props}
+            ref={composedRefs}
+          />
+          {children}
+        </div>
+      </InputContext.Provider>
+    );
+  },
+);
+
+type SlotProps = React.ComponentPropsWithoutRef<typeof RadixSlot> & {
+  size?: keyof typeof SIZE.Slot;
+  position?: "leading" | "trailing";
+};
+type SlotRef = React.ElementRef<typeof RadixSlot>;
+
+const Slot = React.forwardRef<SlotRef, SlotProps>(
+  ({ position = "leading", ...props }, forwardedRef) => {
+    const context = React.useContext(InputContext);
+    return (
+      <div
+        className={clsx(
+          "group box-border flex items-center justify-center h-full",
+          "[&>[data-tgph-button]]:w-full [&>[data-tgph-button]]:h-auto",
+          // Overrides to match the icon button in figma
+          "[&>[data-tgph-button-layout='icon-only']]:aspect-square [&>[data-tgph-button-layout='icon-only']]:p-0",
```

Comment:
Would it be better here to pull these out into its own CSS file? I know end result is similar, just thinking about maintainability and some of these tailwind classes get hard to read

Reply:
With scoping and how that's wired through tailwind I'd like to keep all of these through tailwind as much as possible so that it's one less vector for something to go wrong later on. I think I'd be fine with the approach of having the css file if all of the styles were through that, but to mix and match just seems like it could get muddy later on.